ssl/t1_lib.c

```diff
@@ -2870,22 +2870,19 @@ int tls1_check_chain(SSL *s, X509 *x, EVP_PKEY *pk, STACK_OF(X509) *chain,
 
         ca_dn = s->s3.tmp.peer_ca_names;
 
-        if (!sk_X509_NAME_num(ca_dn))
+        if (ca_dn == NULL || sk_X509_NAME_num(ca_dn) == 0)
             rv |= CERT_PKEY_ISSUER_NAME;
-
-        if (!(rv & CERT_PKEY_ISSUER_NAME)) {
-            if (ssl_check_ca_name(ca_dn, x))
-                rv |= CERT_PKEY_ISSUER_NAME;
-        }
-        if (!(rv & CERT_PKEY_ISSUER_NAME)) {
+        else if (ssl_check_ca_name(ca_dn, x))
+            rv |= CERT_PKEY_ISSUER_NAME;
+        else
             for (i = 0; i < sk_X509_num(chain); i++) {
                 X509 *xtmp = sk_X509_value(chain, i);
                 if (ssl_check_ca_name(ca_dn, xtmp)) {
```

Comment:
Blank line above this (separate declarations)?

Reply:
It was there before, but yeah OK.